File: EasyQuestions/src/Arrays/maxConsecutiveOnes.java
package Arrays;
// Got
// Given a binary array nums, return the maximum number of consecutive 1's in the array.
public class maxConsecutiveOnes {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maxCounter = 0;
        int counter = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {

                counter++;
            }
            if (nums[i] == 0 || i == nums.length-1){
                if (counter > maxCounter) {
                    maxCounter = counter;
                }
                counter = 0;

            }
        }
        return maxCounter;
    }
}
// Time complexity : O(n)
// Space complexity : O(1)

File: src/Step3/3.1/max_consecutive_ones.py
'''
Maximum Consecutive Ones
Given a binary array nums, return the maximum number of consecutive 1's in the array.
Example 1:

Input: nums = [1,1,0,1,1,1]
Output: 3
Explanation: The first two digits or the last three digits are consecutive 1s. The maximum number of consecutive 1s is 3.
Example 2:

Input: nums = [1,0,1,1,0,1]
Output: 2
'''


class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxc = 0
        count = 0
        for i in nums:
            if (i == 1):
                count +=1
            else:
                if (maxc < count):
                    maxc = count
                count = 0
        if (maxc < count):
            return count
        return maxc
        
# Time complexity: O(n)